modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -24,12 +24,69 @@ const freefallRtl = (root: SugarElement<Node>): Optional<SugarElement<Node>> =>
   return child.bind(freefallRtl).orThunk(() => Optional.some(root));
 };
 
-const emptyCells = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[]) => {
+const removeContentFromCells = (cells: SugarElement<HTMLTableCellElement>[]) =>
   Arr.each(cells, PaddingBr.fillWithPaddingBr);
-  editor.selection.setCursorLocation(cells[0].dom, 0);
+
+const getOutsideBlock = (editor: Editor, container: Node): Optional<SugarElement<HTMLElement>> =>
+  Optional.from(editor.dom.getParent(container, editor.dom.isBlock) as HTMLElement).map(SugarElement.fromDom);
+
+const handleEmptyBlock = (editor: Editor, startInTable: boolean, emptyBlock: Optional<SugarElement<HTMLElement>>) => {
+  emptyBlock.each((block) => {
+    if (startInTable) {
+      // Note that we don't need to set the selection as it'll be within the table
+      Remove.remove(block);
+    } else {
+      // Set the cursor location as it'll move when filling with padding
+      PaddingBr.fillWithPaddingBr(block);
+      editor.selection.setCursorLocation(block.dom, 0);
+    }
+  });
+};
+
+/*
+ * Runs when
+ * - the start and end of the selection is contained within the same table (called directly from deleteRange)
+ * - part of a table and content outside is selected
+ */
+const emptySingleTableCells = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[], outsideDetails: Optional<TableDeleteAction.OutsideTableDetails>) => {
+  // Remove content from selected cells
+  removeContentFromCells(cells);
+
+  // Delete all content outside of the table that is in the selection
+  outsideDetails.map(({ rng, isStartInTable }) => {
+    // Get the outside block before deleting the contents
+    const outsideBlock = getOutsideBlock(editor, isStartInTable ? rng.endContainer : rng.startContainer);
+    rng.deleteContents();
+    // Handle block outside the table if it is empty since rng.deleteContents leaves it
+    handleEmptyBlock(editor, isStartInTable, outsideBlock.filter(Empty.isEmpty));
+  });
+
+  // Set the selection:
+  // - to the first emptied cell if the selection was previously inside a table
+  // - otherwise just collapse the previous selection that was in the outside block
+  const selection = editor.selection;
+  if (outsideDetails.forall(({ isStartInTable }) => isStartInTable)) {
```

Comment:
```suggestion
  if (outsideDetails.forall((details) => details.isStartInTable)) {
```
I know we just wrote this, but looking back I think this would be easier to read.